EditableIntegerPicker/tests/EditableIntegerPicker-specs.js

```diff
@@ -1,258 +1,269 @@
-			input.value = 138;
-			input.blur();
+		userEvent.click(screen.getByText('12'));
 
-			picker.update();
+		const input = screen.getByLabelText('undefined change a value with left right button').children.item(1);
+		const inputField = input.children.item(0).children.item(0);
 
-			const expected = 0;
-			const actual = handleChange.mock.calls.length;
+		fireEvent.focus(inputField);
+		userEvent.type(inputField, 'invalid');
+		fireEvent.blur(inputField);
 
-			node.parentNode.removeChild(node);
+		expect(handleChange).not.toHaveBeenCalled();
+	});
 
-			expect(actual).toBe(expected);
-		}
-	);
+	test('should not send two change event when incrementing from edit mode', () => {
+		const handleChange = jest.fn();
+		render(
+			<EditableIntegerPicker
+				defaultValue={15}
+				max={100}
+				min={0}
+				noAnimation
+				onChange={handleChange}
+				step={1}
+			/>
+		);
 
-	test(
-		'should not send two change event when incrementing from edit mode',
-		() => {
-			const handleChange = jest.fn();
-			const node = document.body.appendChild(document.createElement('div'));
-			const picker = mount(
-				<EditableIntegerPicker onChange={handleChange} min={0} max={100} defaultValue={10} step={1} noAnimation />,
-				{attachTo: node}
-			);
+		userEvent.click(screen.getByText('15'));
 
-			picker.find('PickerItem').simulate('click', {target: {type: 'click'}});
+		const input = screen.getByLabelText('undefined change a value with left right button').children.item(1);
+		const inputField = input.children.item(0).children.item(0);
 
-			const input = node.querySelector('input');
-			picker.find('input').simulate('focus');
-			input.value = 38;
-			increment(picker);
-			input.blur();	// It seems it must be manually blurred
+		fireEvent.focus(inputField);
+		userEvent.type(inputField, '12');
+		fireEvent.blur(inputField);
 
-			picker.update();
+		const expected = 1;
 
-			const expected = 1;
-			const actual = handleChange.mock.calls.length;
+		expect(handleChange).toHaveBeenCalledTimes(expected);
+	});
 
-			node.parentNode.removeChild(node);
+	test('should enable input field when some number is typed on the picker', () => {
+		render(<EditableIntegerPicker defaultValue={10} max={100} min={0} step={1} />);
+		const input = screen.getByText('10');
 
-			expect(actual).toBe(expected);
-		}
-	);
+		fireEvent.keyDown(input, {keyCode: 50});
 
-	test(
-		'should enable input field when some number is typed on the picker',
-		() => {
-			const picker = mount(
-				<EditableIntegerPicker min={0} max={100} defaultValue={10} step={1} />
-			);
+		setTimeout((done) => {
+			const expected = 'input';
+			const actual = screen.getByLabelText('undefined change a value with left right button').children.item(1);
 
-			picker.simulate('keyDown', {keyCode: 50});
-			const expected = 1;
-			const actual = picker.find('input').length;
-
-			expect(actual).toBe(expected);
-		}
-	);
+			expect(actual).toHaveClass(expected);
+			done();
```

Comment:
here again. the code inside settimeout is never called. Maybe Mikyung's solution can be applied here

Reply:
Changed them to work properly, thanks for the catch.